### PR TITLE
Improve Client Credential automation test cases

### PR DESCRIPTION
**Pull request by Thumimku**
Public Issue : https://github.com/wso2/product-is/issues/21313

Current CC grant test case only test happy path, with this we are testing following error scenarios along with JWT type token

  - InvalidClientId
  - InvalidClientCredentials
  - WithoutCredentials

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceClientCredentialTestCase.java [hunk 1]
```diff
@@ -130,13 +159,14 @@ public void testGetTokenUsingClientCredentialsGrant() throws Exception {
...
-        Scope scope = new Scope(OAUTH2_SCOPE_OPENID, "xyz", VALID_RANDOM_SCOPE);
+        Scope scope = new Scope(OAUTH2_SCOPE_OPENID, "xyz", VALID_RANDOM_SCOPE, INTERNAL_BULK_RESOURCE_CREATE);
 
         URI tokenEndpoint = new URI(getTenantQualifiedURL(OAuth2Constant.ACCESS_TOKEN_ENDPOINT, tenantInfo.getDomain()));
         TokenRequest request = new TokenRequest(tokenEndpoint, clientAuth, clientCredentialsGrant, scope);
         HTTPResponse tokenHTTPResp = request.toHTTPRequest().send();
         Assert.assertNotNull(tokenHTTPResp, "Access token http response is null.");
-
+        Assert.assertEquals(tokenHTTPResp.getContentType().toString(), "application/json",
+                "Token response did not indicate correct token response");
```
Comment (asha15) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceClientCredentialTestCase.java:169]:
```suggestion
                "Token response did not indicate correct token response.");
```

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceClientCredentialTestCase.java [hunk 2]
```diff
@@ -170,4 +205,114 @@ public void testValidateAccessToken() throws Exception {
...
+        Assert.assertEquals(errorObject.getDescription(),
+                "Client credentials are invalid.",
+                "Invalid access token response doesn't contain required error description.");
+    }
+
+    @Test(groups = "wso2.is", description = "Send client credentials token request without client secret.",
+            dependsOnMethods = "testGetTokenUsingCCGrantWithInvalidClientCredentials")
+    public void testGetTokenUsingCCGrantWithoutCredentials() throws Exception {
+
+
```
Comment (asha15) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceClientCredentialTestCase.java:273]: Remove the additional line.

Comment (Thumimku) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceClientCredentialTestCase.java]: fixed with https://github.com/wso2/product-is/pull/21449/commits/282d1fdb1dce22ea051a343452d7d8f3436b059f

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceClientCredentialTestCase.java (hunk 1 above)
Comment (Thumimku) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceClientCredentialTestCase.java]: fixed with https://github.com/wso2/product-is/pull/21449/commits/b171a8e17cd19f4b327da73e8192945bc0e79a78

**Review by asha15 — APPROVED** [review]

**Pull request merged**